airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py

```diff
@@ -21,70 +21,222 @@
+            if not all([current_state, latest_state]):
+                max_value = list(filter(bool, [current_state, latest_state]))[0]
+            else:
+                max_value = max(latest_state, current_state)
+            output = {self.campaign_id: {self.cursor_field: max_value}}
+            return output
+
+        current_campaign_state = current_state.get(self.cursor_field) if current_state else None
+        latest_campaign_state = latest_state.get(self.cursor_field) if latest_state else None
+        if not all([current_campaign_state, latest_campaign_state]):
+            nonzero_state = list(filter(bool, [current_campaign_state, latest_campaign_state]))[0]
+            output = {self.campaign_id: {self.cursor_field: nonzero_state}}
+        else:
+            max_value = max(current_campaign_state, latest_campaign_state)
+            output = {self.campaign_id: {self.cursor_field: max_value}}
+        return output
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = MailChimpStream.request_params(self, stream_state=stream_state, **kwargs)
+
+        since_value_camp = stream_state.get(self.campaign_id)
+        if since_value_camp:
+            since_value = since_value_camp.get(self.cursor_field)
+            if since_value:
+                params["since"] = since_value
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        # transform before save
+        # [{'campaign_id', 'list_id', 'list_is_active', 'email_id', 'email_address', 'activity[array[object]]', '_links'}] ->
+        # -> [[{'campaign_id', 'list_id', 'list_is_active', 'email_id', 'email_address', '**activity[i]', '_links'}, ...]]
+        data = response_json[self.data_field]
+        for item in data:
+            for activity_record in item["activity"]:
+                new_record = {k: v for k, v in item.items() if k != "activity"}
+                for k, v in activity_record.items():
+                    new_record[k] = v
+                yield new_record
+
+
+class HttpBasicAuthenticator(HttpAuthenticator):  # should not be placed in this file but we havent it!
+    def __init__(self, auth: Tuple[str, str]):
+        self.auth = auth
+
+    def get_auth_header(self) -> Mapping[str, str]:
+        # origin: resp = requests.get(url, auth=(anystring, 'your_apikey'))
+        # -> use Basic Authorization header instead
+        auth_string = f"{self.auth[0]}:{self.auth[1]}".encode("utf8")
+        b64_encoded = base64.b64encode(auth_string).decode("utf8")
+        return {"Authorization": "Basic " + b64_encoded}
+
+
+class SourceMailchimp(AbstractSource):
+    def _setup_properties(self, config: Mapping[str, Any]) -> None:
+        if not hasattr(self, "_client"):
+            self._client = MailChimp(mc_api=config["apikey"], mc_user=config["username"])
```

Comment:
do we always need to create MailChimp client? if not I think the value of this method is questionable, WDYT?